pkg/cluster/cluster.go

```diff
@@ -129,6 +150,9 @@ func (c *Cluster) forSpecificMachines(do func(*Machine, int) error, machineNames
 }
 
 func (c *Cluster) ensureSSHKey() error {
+	if c.spec.Cluster.PrivateKey == "" {
```

Comment:
This breaks my use case. I do the following:
- `CreatePublicKey(name: "my-key.pub", key: "ssh-rsa ...")`
- `CreateCluster(name: "my-cluster", privateKey: "my-key")`
- `CreateMachine(cluster: "my-cluster", name: "machine1")`

After the commands above, I was expecting being able to ssh to `machine1` using the private key matching `my-key.pub` uploaded via `CreatePublicKey`. This fails because apparently this function overwrites the public key that I passed in `CreatePublicKey`.

Reply:
Gah! indeed, using the same name in a `CreatePublicKey` and the cluster key will interact badly. They are two different namespaces. An easy way out is to separate put those two namespaces in two different directories so they don't override each other. Will fix!